docs/architecture/decisions/adr-004.md

```diff
@@ -0,0 +1,33 @@
+[accessibility support baseline]: https://www.w3.org/TR/WCAG-EM/#step1c
+
+# ADR 4: Model browser specific values and computations
+
+## Context
+
+Having decided on [ADR 2](adr-002.md), we will no longer be tied to any one specific browser and are left with the choice of what to do about things that browsers are not aligned on. For example, if Alfa needs to implement an API to perform a given task and a given set of browsers do not agree on how the API should be implemented, Alfa can do several things:
+
+1. Implement the API the way it is implemented in the _least_ capable browser, in terms of the API, among a given set of browsers.
+
+2. Implement the API the way it is implemented in the _most_ capable browser, in terms of the API, among a given set of browsers.
+
+3. Provide several implementations of the API that cover all the different implementations of the API among a given set of browsers.
+
+Our proprietary accessibility conformance testing engine at Siteimprove takes the second approach in practice as it is usually run within the latest version of the Chrome browser and relies heavily on browser APIs. Other tools that operate with the concept of an [accessibility support baseline][] will typically take a combination of the first and second approach where some browser APIs are used and then others implemented manually according to the capabilities of the [accessibility support baseline][].
```

Comment:
@singingknight said in the comments that 

> A fourth possibility which you could argue is the one used in the old engine is:
Implement the API the way it is implemented in the _most_ commonly used browser among a given set of browsers.
This may be the same as either 1 or 2, but might be a third way.

Just wondering about inconsistency here. We should all agree on how Alfa relates to the proprietary engine.